### PR TITLE
NH-24134: upgrade the redis instrumentation for redis-rb > 5

### PR DESCRIPTION
**Pull request by xuan-cao-swi**
## Why?
redis-rb refactored code structure from version 5 so the instrumentation and test need to be updated for this apm.

## Impact?
1. Redis test case: separated the redis test case from instrumentation
2. Keep redis 4 compatibility

## Changes

For single call, the trace content is same except all the digital number is now string (e.g. 1 becomes "1").
For pipeline call, the trace changed because there is no pipeline call anymore; even it's pipeline call, it will be dividing to multiple single call by redis (by design). [Reference](https://github.com/redis/redis-rb/blob/master/CHANGELOG.md#500)

Separation of redis test to rest of others since redis sometimes is hard to manage. Minitest skip is not working as expected, so separate them are better for test management.

Some of the test cases will work for redis 4.x but not all of them. Currently the test is not running against 4.x since the gemfile we have `gem 'redis'` without indicating the version so it will always be the latest (5.x). However, user can still instrument redis 4.x since we have preserved `redis_v4.rb` for now.
 
More [info](https://swicloud.atlassian.net/wiki/spaces/NIT/pages/3332442261/Redis+Instrumentation) on confluence page.

### CODE REVIEW

**Review by cheempz — COMMENTED** [review]
@xuan-cao-swi congrats on adding your first instrumentation! I have a few asks:
* can you showcase (either in this PR, or in the Jira or a Confluence page) the before and after of this feature? i.e. before this change, links to traces of redis client 4.x and 5.x, and compared to traces from after this change. 
* give more background why the redis gemfile and tests were split out
* are tests still being run against redis 4.x?
* the GHA [tests on ruby 3.1](https://github.com/solarwindscloud/solarwinds-apm-ruby/actions/runs/3518784074/jobs/5899390098) logged these type of ERRORs several times:
> E, [2022-11-21T15:16:42.354090 #12328] ERROR -- : [solarwinds_apm/loading] Error loading instrumentation file '/__w/solarwinds-apm-ruby/solarwinds-apm-ruby/lib/solarwinds_apm/inst/redis.rb' : uninitialized constant Redis

>  if defined?(::RedisClient) && Gem::Version.new(Redis::VERSION) >= Gem::Version.new('5.0.0')

and there were several of these
> /__w/solarwinds-apm-ruby/solarwinds-apm-ruby/lib/solarwinds_apm/inst/redis.rb:9: warning: already initialized constant SolarWindsAPM::Inst::Redis::Client::NO_KEY_OPS

Could you investigate?

**Comment by xuan-cao-swi** [comment]
> @xuan-cao-swi congrats on adding your first instrumentation! I have a few asks:
> 
> * can you showcase (either in this PR, or in the Jira or a Confluence page) the before and after of this feature? i.e. before this change, links to traces of redis client 4.x and 5.x, and compared to traces from after this change.
> * give more background why the redis gemfile and tests were split out
> * are tests still being run against redis 4.x?
> * the GHA [tests on ruby 3.1](https://github.com/solarwindscloud/solarwinds-apm-ruby/actions/runs/3518784074/jobs/5899390098) logged these type of ERRORs several times:
> 
> > E, [2022-11-21T15:16:42.354090 #12328] ERROR -- : [solarwinds_apm/loading] Error loading instrumentation file '/__w/solarwinds-apm-ruby/solarwinds-apm-ruby/lib/solarwinds_apm/inst/redis.rb' : uninitialized constant Redis
> 
> > if defined?(::RedisClient) && Gem::Version.new(Redis::VERSION) >= Gem::Version.new('5.0.0')
> 
> and there were several of these
> 
> > /__w/solarwinds-apm-ruby/solarwinds-apm-ruby/lib/solarwinds_apm/inst/redis.rb:9: warning: already initialized constant SolarWindsAPM::Inst::Redis::Client::NO_KEY_OPS
> 
> Could you investigate?

Thanks for the review.
After investigation, it seems like the rails70.gemfiles have sidekiq only require redis-client gem but no redis gem so that error pop up. (fixed)

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,17 @@ https://github.com/solarwindscloud/solarwinds-apm-ruby/releases
 
 Dates in this file are in the format MM/DD/YYYY.
 
+# solarwinds_apm 5.1.4 (11/23/2022)
```
Comment (cheempz) [inline CHANGELOG.md:6]: Reminder: update to actual release date when new gem goes out.

**Review by cheempz — APPROVED** [review]
LGTM, thanks for the revisit @xuan-cao-swi!

File: CHANGELOG.md (hunk 1 above)
Comment (xuan-cao-swi) [reply CHANGELOG.md:6]: well do

**Pull request merged**